doc/development-setup.md

```diff
@@ -0,0 +1,24 @@
+# Set up Develop Environment
```

Comment:
should envoy development link be added too? Once users have that.. all of the things will follow?

Reply:
No, we should actually avoid that. It is confusing (Wasm extension development should be independent from the proxy) and not necessary (Envoy development set up is more complex). Oncehttps://github.com/proxy-wasm/proxy-wasm-cpp-sdk/pull/71 is in, we should point to [instruction there](https://github.com/proxy-wasm/proxy-wasm-cpp-sdk/pull/71/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R86) for development set up.